sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -357,3 +361,126 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_missing_indicator():
+    X1_orig = np.array([
+        [np.nan,  np.nan,   1,   3],
+        [4,  np.nan,   0,  np.nan],
+        [8,  np.nan,   1,  0],
+        [0,  np.nan,   0,  15],
+        [16,  np.nan,   1,  19]
+    ])
+    X2_orig = np.array([
+        [5,  1,   1,   np.nan],
+        [np.nan,  np.nan,   2,  3],
+        [2,  3,   4,  0],
+        [0,  np.nan,   5,  np.nan],
+        [11,  np.nan,   1,  1]
+    ])
+
+    def assert_type(actual, X_is_sparse, sparse_param, missing_values):
+        if sparse_param is True:
+            assert_equal(actual, sparse.csc_matrix)
+        elif ((sparse_param == "auto" and missing_values == 0) or
+              sparse_param is False):
+            assert_equal(actual, np.ndarray)
+        else:
+            if X_is_sparse:
+                assert_equal(actual, sparse.csc_matrix)
+            else:
+                assert_equal(actual, np.ndarray)
+
+    def assert_mask(actual, expected, features):
+        if len(features) != 0:
```

Comment:
I think the code below would be more readable if you dropped `features` from `assert_mask` and just did `[:, features]` when calling below.